crates/trie/sparse/src/trie.rs

```diff
@@ -969,24 +969,6 @@ impl SparseTrieInterface for SerialSparseTrie {
         full_path: &Nibbles,
         expected_value: Option<&Vec<u8>>,
     ) -> Result<LeafLookup, LeafLookupError> {
-        // Helper function to check if a value matches the expected value
-        fn check_value_match(
```

Comment:
can we just add an `#[inline]` attribute here? It doesn't clone the value unless the error is returned anyway, as @mediocregopher said.